File: src/main/java/practice/account/domain/TransactionHistory.java
package practice.account.domain;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.experimental.FieldDefaults;

@Getter
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class TransactionHistory {

  Long id;
  LocalDateTime transactionDateTime;
  TransactionType type;
  BigDecimal amount;
  String counterPartyBank;
  String counterPartyAccountId;
  Long paymentId;

  public LocalDate getTransactionDate() {
    return transactionDateTime.toLocalDate();
  }

  public void setToDeposit() {
    this.type = TransactionType.DEPOSIT;
  }

}

File: src/main/kotlin/practice/account/domain/TransactionHistory.kt
package practice.account.domain

import java.math.BigDecimal
import java.time.LocalDate
import java.time.LocalDateTime

class TransactionHistory(
    var id: Long? = null,
    var transactionDateTime: LocalDateTime? = null,
    var type: TransactionType? = null,
    var amount: BigDecimal? = null,
    var counterPartyBank: String? = null,
    var counterPartyAccountId: String? = null,
    var paymentId: Long? = null
) {

    fun transactionDate(): LocalDate? {
        return transactionDateTime?.toLocalDate()
    }

    fun setToDeposit() {
        this.type = TransactionType.DEPOSIT
    }
}
